src/domain/organization/organizationForm/merchants/Merchants.tsx

```diff
@@ -0,0 +1,84 @@
+import { FieldArray, useField } from 'formik';
+import { Button, IconPlus } from 'hds-react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import Notification from '../../../../common/components/notification/Notification';
+import { OrganizationFieldsFragment } from '../../../../generated/graphql';
+import FormRow from '../../../admin/layout/formRow/FormRow';
+import FieldWithButton from '../../../event/layout/FieldWithButton';
+import useUser from '../../../user/hooks/useUser';
+import {
+  ORGANIZATION_FIELDS,
+  WEB_STORE_MERCHANT_INITIAL_VALUES,
+} from '../../constants';
+import { WebStoreMerchantFormFields } from '../../types';
+import { checkIsEditMerchantAllowed } from '../../utils';
+import Merchant from './merchant/Merchant';
+
+interface Props {
+  isEditingAllowed: boolean;
+  organization?: OrganizationFieldsFragment;
+}
+
+const getMerchantPath = (index: number) =>
+  `${ORGANIZATION_FIELDS.WEB_STORE_MERCHANTS}[${index}]`;
+
+const Merchants: React.FC<Props> = ({ isEditingAllowed, organization }) => {
+  const { t } = useTranslation();
+  const { user } = useUser();
+  const { editable, warning } = checkIsEditMerchantAllowed({ t, user });
+
+  const [{ value: webStoreMerchants }] = useField<WebStoreMerchantFormFields[]>(
+    {
+      name: ORGANIZATION_FIELDS.WEB_STORE_MERCHANTS,
+    }
+  );
+
+  return (
+    <FieldArray
+      name={ORGANIZATION_FIELDS.WEB_STORE_MERCHANTS}
+      render={(arrayHelpers) => (
+        <>
+          {!!warning && (
+            <FormRow>
+              <Notification type="info" label={warning} />
+            </FormRow>
+          )}
+          {webStoreMerchants.map((merchant, index) => {
+            return (
+              <Merchant
+                key={index}
```

Comment:
Just a note that SonarCloud is saying not to use `index` as the `key`, but I think this is acceptable since there is only going to be at most one merchant in this list.

Reply:
This is intentional. Using anything else but index as a key causes some issues with Formik FieldArray. So to make formik to work as expected I have used the index as a key even it's not the best practise.